python/src/robyn/modeling/clustering/cluster_builder.py

```diff
@@ -0,0 +1,167 @@
+        """
+        pass
+
+    def calculate_confidence_intervals(
+        self, config: ClusteringConfig
+    ) -> Dict[str, float]:
+        """
+        Calculates bootstrapped confidence intervals for model performance metrics.
+
+        Args:
+            config (ClusteringConfig): Configuration for the clustering process.
+
+        Returns:
+            Dict[str, float]: A dictionary containing confidence intervals for each model.
+        """
+        pass
+
+    def compute_error_scores(
+        self, weights: List[float], config: ClusteringConfig
+    ) -> pd.DataFrame:
+        """
+        Computes error scores for the models based on specified performance metrics.
+
+        Args:
+            weights (List[float]): A list of weights for the error metrics.
+            config (ClusteringConfig): Configuration for the clustering process.
+
+        Returns:
+            pd.DataFrame: A DataFrame containing the computed error scores for each model.
+        """
+        pass
+
+    def select_optimal_clusters(
+        self, df: pd.DataFrame, config: ClusteringConfig
+    ) -> int:
+        """
+        Selects the optimal number of clusters based on WSS variance.
+
+        Args:
+            df (pd.DataFrame): The prepared data for clustering.
+            config (ClusteringConfig): Configuration for the clustering process.
+
+        Returns:
+            int: The optimal number of clusters.
+        """
+        pass
+
+    def prepare_data_for_clustering(self, config: ClusteringConfig) -> pd.DataFrame:
+        """
+        Prepares the data frame for clustering based on specified criteria.
+
+        Args:
+            config (ClusteringConfig): Configuration for the clustering process.
+
+        Returns:
+            pd.DataFrame: A DataFrame prepared for clustering.
+        """
+        pass
+
+    def normalize_values(
+        self, values: List[float], min_val: float = 0, max_val: float = 1
+    ) -> List[float]:
+        """
+        Normalizes a list of values to a specified range.
+
+        Args:
+            values (List[float]): The values to normalize.
+            min_val (float): The minimum value of the normalized range.
+            max_val (float): The maximum value of the normalized range.
+
+        Returns:
+            List[float]: A list of normalized values.
+        """
+        pass
+
+    def select_top_models(
+        self, clustered_df: pd.DataFrame, config: ClusteringConfig
+    ) -> pd.DataFrame:
+        """
+        Selects the top models based on their distance to the origin.
+
+        Args:
+            clustered_df (pd.DataFrame): The DataFrame containing cluster assignments.
+            config (ClusteringConfig): Configuration for the clustering process.
+
+        Returns:
+            pd.DataFrame: DataFrame of top models.
+        """
+        pass
+
+    def plot_confidence_intervals(
```

Comment:
let's take out plot from cluster component here. I see some of the methods here already located at cluster_visualization.py https://github.com/facebookexperimental/Robyn/blob/robynpy_release/python/src/robyn/visualization/cluster_visualization.py